server/src/test/java/org/elasticsearch/action/bulk/TransportShardBulkActionTests.java

```diff
@@ -1072,6 +1091,70 @@ public void testPerformOnPrimaryReportsBulkStats() throws Exception {
         latch.await();
     }
 
+    public void testNoopMappingUpdateInfiniteLoopPrevention() throws Exception {
```

Comment:
I wonder if this test tests a non-existing scenario?

The noop prevention would only concern a case where the mapping used during parsing is different from the mapping used during preflight, containing the mapping changes added. And if so, we'd expect any retry to always succeed?

Reply:
> I wonder if this test tests a non-existing scenario?

I hope it does :)

This is meant as a safety net in case there's an unexpected bug that lead to a situation where we accept a field in `DocumentParserContext` even though it turns out that don't  have enough capacity for it when merging the mappings.